### DISCUSSION TITLE
Slide Show

### DISCUSSION THREAD

**Discussion by starpause** [koreader/koreader#9968] [Ideas]
I'd like to use my old eReaders as a digital art displays. When starting the slideshow a directory of images would be displayed. Nice to have features would be:

- Setting the length of time before switching images
- Backlight settings (brightness, temperature)
- Display order (random, sequential)
- Crop behavior (fill, fit)

I explored the option of using the screensaver option, however screensaver always turns the backlight off (as the screensaver is displayed when the device goes to sleep). I also tried the native "demo mode on" behavior for Kobos but couldn't get an image to display (maybe I have the images in the wrong directory, I'd just get a white screen and no image slide show).

**Comment by NiLuJe** [comment]
Sidebar: IIRC, the Kobo demo mode works like the boot animation, in that it expects a gzipped framebuffer dump at whatever insane framebuffer layout it is at boot (probably the rotated 16bpp insanity).

----

Sounds like a job for a small plugin w/ ImageViewer (or maybe just ImageWidget, since you don't need any controls).

Or a DIY thingy w/ FBInk.

**Reply by starpause** [reply]
[ImageWidget](https://koreader.rocks/doc/modules/ui.widget.imagewidget.html) seems fitting. Since it would be a KOreader feature I think that makes more sense as a starting point than [FBInk](https://github.com/NiLuJe/FBInk) (but starred that repo as well, thanks for all your work on these!).

**Reply by poire-z** [reply]
Using KOReader facilities to build these little toys is probably the easiest and best thing to do.
But it will be _your user plugin_ :) and not a KOReader feature, as it's a reader and not a digital art display/picture frame :) (my take on that at https://github.com/koreader/koreader/pull/5854#issuecomment-586579954)
